tf2onnx/tfonnx.py

```diff
@@ -1745,12 +1745,116 @@ def sparse_softmax_cross_entropy_with_logits_op(ctx, node, name, args):
+    body_nodes = list([slice_node.op, new_line.op])
+    body_nodes.append(utils.make_onnx_identity("cond", "cond_out"))
+    body_nodes.append(utils.make_onnx_identity("line", "res"))
+    body_graph = helper.make_graph(body_nodes, utils.make_name("MatrixBandPart_body"), body_ins, body_outs)
+    # initial value of body vars
+    shape = ctx.make_node(op_type="Shape", inputs=[node.input[0]])  # dtype of result is int64
+    node_name = utils.make_name("line_num_index")
+    col_or_row_num_index = ctx.make_const(name=node_name, np_val=np.array(axis).astype(np.int32))
+    line_num = ctx.make_node(op_type="Gather", inputs=[shape.output[0], col_or_row_num_index.output[0]])
+
+    trip_cnt = line_num.output[0]
+    cond = ctx.make_const(name="true", np_val=np.array(1).astype(np.bool)).output[0]
+    col_init = one_line.output[0]
+
+    loop_node = ctx.make_node(op_type="Loop", inputs=[trip_cnt, cond, col_init], output_count=2,
+                              attr={"body": body_graph})
+    # convert generated mask matrix from bool to right shape and data type
+    squeeze = ctx.make_node(op_type="Squeeze", inputs=[loop_node.output[1]], attr={"axes": [squeeze_axis]})
+    cast1 = ctx.make_node(op_type="Cast", inputs=squeeze.output, attr={"to": onnx_pb.TensorProto.FLOAT})
+    mask_matrix = ctx.make_node(op_type="Transpose", inputs=cast1.output) if axis == 1 else squeeze
+    cast2 = ctx.make_node(op_type="Cast", inputs=mask_matrix.output,
+                          attr={"to": ctx.get_dtype(node.input[0])})
+    res = ctx.make_node(op_type="Mul", inputs=[cast2.output[0], node.input[0]], outputs=node.output)
+    return [first_col_or_row, first_col_or_row_casted, zero_line, one_line, shape,
+            line_num, loop_node, squeeze, cast1, mask_matrix, cast2, res]
+
+
+def floordiv_op(ctx, node, name, args):
+    res = [node]
+    node.type = "Div"
+    dtype = ctx.get_dtype(node.input[0])
+    if dtype in [onnx_pb.TensorProto.FLOAT, onnx_pb.TensorProto.FLOAT16, onnx_pb.TensorProto.DOUBLE]:
+        floor_res = ctx.insert_new_node_on_output(op_type="Floor", output_name=node.output[0], name="floor_div_res")
+        res.append(floor_res)
+    return res
+
+
+def floormod_op(ctx, node, name, args):
+    nodes = []
+    div = ctx.make_node(op_type="Div", inputs=node.input)
+    nodes.append(div)
+    dtype = ctx.get_dtype(node.input[0])
+    if dtype in [onnx_pb.TensorProto.FLOAT, onnx_pb.TensorProto.FLOAT16, onnx_pb.TensorProto.DOUBLE]:
+        div = ctx.make_node(op_type="Floor", inputs=div.output)
+        nodes.append(div)
+    mul = ctx.make_node(op_type="Mul", inputs=[div.output[0], node.input[1]])
+    res = ctx.make_node(op_type="Sub", inputs=[node.input[0], mul.output[0]], outputs=node.output)
+    nodes.extend([mul, res])
+    return nodes
+
+
+def all_op(ctx, node, name, args):
+    cast = ctx.make_node(op_type="Cast", inputs=[node.input[0]], attr={"to": onnx_pb.TensorProto.INT32})
+    reduce_dim = node.inputs[1].get_tensor_value()
```

Comment:
not always in tensorflow while in onnx is required since reduce_dim is specified in onnx attribute.

and get_tensor_value will help us to throw an exception.